### ISSUE TITLE
Windows Postman App - High CPU usage latest upgrade v6.0.10

### ISSUE THREAD

**Issue by vinodpy** [a85/POSTMan-Chrome-Extension#4361]
I recently upgraded to version v6.0.10 on Windows Postman App and the idle CPU usage for Postman App is hovering between 75-90%. There is no request going on at the time of high CPU usage.
## App Details:

```
Postman for Windows
Version 6.0.10
win32 6.3.9600 / x64
```

## Issue Report:
 1. Did you encounter this recently, or has this bug always been there: Recently after upgrade to v6.0.10
 2. Expected behaviour: CPU usage not so high
 3. Console logs (http://blog.getpostman.com/2014/01/27/enabling-chrome-developer-tools-inside-postman/ for the Chrome App, View->Toggle Dev Tools for the Mac app):
 4. Screenshots (if applicable)

**Comment by vinodpy** [comment]
Here are things I have tried til now
1. Did clean reinstall of v6.0.10
2. Installed the previous v6.0.9 and I did see the same issue(before upgrading to v6.0.10 I am not sure what was my version )
Appreciate any help on this. I had to switch to Swagger to make my requests for now

**Comment by vinodpy** [comment]
Any update on this?

**Comment by prashantagarwal** [comment]
Hey @vinodpy ! Can you help me by providing some screenshots and logs. 
1) Open devtools
2) Got to **View** -> **Developer** -> Show Devtools (Current View) .
3) In devtools go to **Timeline** tab and click on circle on the top left.
4) Stop after 6-7 sec.

<img width="674" alt="screen shot 2018-03-29 at 9 40 25 pm" src="https://user-images.githubusercontent.com/3762808/38100109-182aa7fe-339a-11e8-940e-feb445486355.png">

Repeat the same steps for **Show Devtools (Shared)**

**Comment by duanefields** [comment]
I see the same behavior on the mac - 95% CPU while doing nothing
<img width="912" alt="screenshot_0055" src="https://user-images.githubusercontent.com/548763/40188271-35c3b33c-59bf-11e8-8e96-bb079a8601c8.png">
<img width="900" alt="screenshot_0054" src="https://user-images.githubusercontent.com/548763/40188276-3835ce7a-59bf-11e8-8bbf-7575c65bc164.png">

**Comment by eduardboamba** [comment]
Guys, this amazing tool has become useless in the last couple of weeks. It spikes up to 100% CPU and >2GB of memory usage when I'm switching between request tabs or just trying to close some of them. I'm setting it aside until the issue is fixed.
Here are some screenshots, in case you find use in them.

![inkedtimeline1_li](https://user-images.githubusercontent.com/11644755/40413450-22f444cc-5e76-11e8-99c4-e797fa619f25.jpg)

I was not quick enough to grab one with 100% CPU usage:
- 01
![taskman01](https://user-images.githubusercontent.com/11644755/40413981-74eac26e-5e77-11e8-9ba0-640a0fb88218.png)

- 02
![taskman02](https://user-images.githubusercontent.com/11644755/40413982-75043046-5e77-11e8-84ea-dcf06ee5a18c.png)

- 03
![taskman03](https://user-images.githubusercontent.com/11644755/40413984-7521ae96-5e77-11e8-80a8-df1df19c92bf.png)

Please find time to look into this defect, it's a huge blocker.

**Comment by deivide** [comment]
Same here:
![image](https://user-images.githubusercontent.com/905281/40843855-bbbc6018-6588-11e8-9c9f-ee5d1040d8e9.png)

**Comment by CaffeinatedTester** [comment]
Can also confirm since upgrade I am having major CPU usage issues when idle. Preventing me from using the tool on my mac.

<img width="549" alt="screen shot 2018-06-02 at 11 39 44" src="https://user-images.githubusercontent.com/15998340/40873537-bc964e46-6659-11e8-85d9-1fafc8bd6e36.png">

**Comment by tdcbm** [comment]
Same issue here in version 6.1.4 and canary also happen too

**Comment by nivle** [comment]
I had kind of the same issue? though for me it just spiked the cpu to 70+% when I started it, 
causing it to kinda break, and take forever to start.
I just enabled compability to be set to windows 8(since I use windows 8) and it fixed that problem.

![image](https://user-images.githubusercontent.com/16451772/49332211-3eff5880-f5a9-11e8-8746-63cd6db29eda.png)

**Comment by a85** [comment]
Everyone, thank you for your patience and your comments. We are tracking performance issue aggressively. If you can give us some data points on the sizes of requests/responses you were dealing with, number of collections in the workspace, open request tabs etc. or any other relevant details, it'll help us set up relevant benchmarks to test against. I'd like to emphasize that performance and a smooth experience is a top priority and your help would be great.

**Comment by bass3l** [comment]
Hey @a85 ,

Thank you guys for this amazing tool. Here are some info so it can help you diagnose the issue:
Version: 7.1.1
Windows 10
Number of collections: 4 with around 20 small requests total
CPU usage while idle: +70% CPU usage at 100% the time
Here is a screenshot:

![Capture](https://user-images.githubusercontent.com/9248745/59883408-aa7e0b80-93bd-11e9-8298-e92b8677adb3.PNG)

Tell me if I can provide any further help. Thanks.

**Comment by jkoehlerrfid** [comment]
I have the same problem (using Win10), but when I start Postman in compatibility mode "Windows 8", it works much better.
Win10 = very high CPU utilization, even 10min after app start
Win10 using compatibility mode "Windows 8" = high CPU utilization in first 10s

**Comment by bothyculture** [comment]
I get this problem with version 7.3.4

**Comment by emircansancar** [comment]
I have that same problem.

![image](https://user-images.githubusercontent.com/16269350/62367032-347ad180-b531-11e9-9527-f7b96d2f85b9.png)

**Comment by eljarete** [comment]
Hello,
Same issue: I haven't got any request running and the application is eating above the 40% of the CPU usage.
![image](https://user-images.githubusercontent.com/38129969/62851979-4d7d4280-bce8-11e9-9a33-e82f00dd5ab2.png)
This makes the system unuseful while developing with postman.

**Comment by angrimer** [comment]
I had this experience as well (since last week), driving me mad! CPU 50%+ when Postman is supposed to idle!

I'm on Postman 7.5.0 on Windows 10 (19H1) with plenty of collections.
I tried uninstalling, reinstalling both 32 and 64 bit versions without success.

HOWEVER, I FOUND A SOLUTION THAT WORKS FOR ME AT LEAST! :-)

The uninstall does not remove the local appdata cache and something in this Postman cache must have gone corrupt or an version upgrade not handling the old cache very well?! After deleting the directories below manually and reinstalling Postman again, everything went back to normal, Postman using 0% CPU when idling, hurray! And that is after logging in again and having restored all my collections from the cloud account.

C:\Users\username\AppData\Local\Postman
C:\Users\username\AppData\Roaming\Postman

Hope this might help somebody.

**Comment by jkirkpatrick** [comment]
Same problem occurring under ubuntu linux.

**Comment by ryparker** [comment]
> I had this experience as well (since last week), driving me mad! CPU 50%+ when Postman is supposed to idle!
> 
> I'm on Postman 7.5.0 on Windows 10 (19H1) with plenty of collections.
> I tried uninstalling, reinstalling both 32 and 64 bit versions without success.
> 
> HOWEVER, I FOUND A SOLUTION THAT WORKS FOR ME AT LEAST! :-)
> 
> The uninstall does not remove the local appdata cache and something in this Postman cache must have gone corrupt or an version upgrade not handling the old cache very well?! After deleting the directories below manually and reinstalling Postman again, everything went back to normal, Postman using 0% CPU when idling, hurray! And that is after logging in again and having restored all my collections from the cloud account.
> 
> C:\Users\username\AppData\Local\Postman
> C:\Users\username\AppData\Roaming\Postman
> 
> Hope this might help somebody.

I have over 8 workspaces containing around 25 collections adding up to over 5000 requests and 3000 prescripts. I was experiencing a 500% CPU usage. However @angrimer 's fix worked and I'm now experiencing little to no interruption.

On mac the folder to delete is located in `~/Library/Application Support/Postman`

Thanks!

**Comment by crunkid** [comment]
Had a similar issue but managed to resolve mine by renaming the old version folder.
I had two versions app-7.9.0 and app-7.10.0, so i renamed app-7.9.0 to app-7.9.0-OLD and now the app works at under 25% CPU Usage.

**Comment by jegsar** [comment]
On windows, same issue and I was able to rename the folder as @crunkid did which seems to help, very odd indeed.

**Comment by belangervince** [comment]
Same issue with version 7.14.0, even after I closed the application, 3 processes were still running and eating RAM and CPU.

**Comment by AnkitPandoh** [comment]
same issue with the latest version

**Comment by jsparksaac** [comment]
Having the same issue here. Started with version 7.14.0 and still occurs with 7.16.0. Occurs on our Windows 10 machines running 1809, but not earlier versions of Windows 10. 

Process Explorer shows cmd.exe launching reg.exe with REG QUERY HKCU\Software\Google\Chrome\NativeMessagingHosts\com.postman.postmanapp over and over again, often pinging the CPU to 100%.

<img width="393" alt="Postman7 14CPU" src="https://user-images.githubusercontent.com/59940235/72539972-ffd7a980-3834-11ea-9df7-90661459e042.PNG">

**Comment by vinitshahdeo** [comment]
@jsparksaac Can you please try installing Interceptor Bridge and see if there is a decrease in CPU consumption?

> Note: You can install Interceptor Bridge either from the Capture Cookies tab or Request tab with source toggled to Interceptor.

<img width="955" alt="Screenshot 2020-01-16 at 10 41 30 PM" src="https://user-images.githubusercontent.com/20594326/72555754-3a882300-38c3-11ea-8674-d5884775f3c3.png">

<img width="850" alt="Screenshot 2020-01-16 at 10 45 34 PM" src="https://user-images.githubusercontent.com/20594326/72555771-496ed580-38c3-11ea-8522-25fb70cc68a5.png">

**Comment by vinitshahdeo** [comment]
@jsparksaac Please try using Postman Canary(`v7.16.0-canary01`). We've shipped a fix to reduce the CPU usage. We're still working on improving performance and hope to have a stable release soon. Please confirm if the Postman Canary app helps to solve the CPU usage issue faced by you.

> **You can download Postman Canary from [here](https://www.getpostman.com/downloads/canary).**

**Comment by buttflattery** [comment]
@vinitshahdeo  i am using up to date! Postman v7.21.1 and the CPU usage remains constant on 80% and shows approx 6 different process for postman even if i have all the request tabs closed and using the compatibility mode windows 8 dropped it back under 10%

**Comment by ABryantdev** [comment]
I had this issue on Windows 10 Postman desktop v7.21.1, was running at 40-60% CPU >1GB RAM.

Fixed by navigating to %AppData%\Postman\Cache (C:\Users\your_user\AppData\Roaming\Postman\Cache) and deleting all files that fit f_* (e.g. f_000001).

I managed to do this with postman open but idle, no re-install required but possibly a restart to the process.

**Comment by LarsScha** [comment]
Same issue here (Postman desktop v7.32.0 on Windows 10). Just opening Postman (without sending any requests) starts 3-4 postman processes of which one just jumps up to 40% CPU usage.

Also, when I close Postman, the process is still there with an equal amount of CPU usage...and sophos antivirus also jumps to 30%-40%. Closing the process manually also drops the antivirus cpu usage.

**Comment by amrbadawy** [comment]
I've the issue, the only way to fix it for me is to follow https://github.com/postmanlabs/postman-app-support/issues/4361#issuecomment-522616177 
Thanks @angrimer

**Comment by satuday** [comment]
Im also experiencing this in Win 10 with Postman v7.30.2-canary01.  As soon as I open postman, taskman show postman is using 30% - 40% cpu.

**Comment by mccannt** [comment]
Thanks for the updates, everyone. We are constantly making improvements in the app and I would advise you to keep up with the latest version to ensure you get these performance updates.

**Comment by ermac0** [comment]
It seems to happen whenever Postman detects new version, then CPU goes to 100%. It won't drop until I click Help -> Check for updates and "Restart and update" then after the app restarts the CPU usage goes back to normal. It remains like that until new version is detected again.
Postman v8.5.1, Windows 10 Ultimate.

**Comment by bteamsoftware** [comment]
This is still an issue.  A few days ago postman did an update.  This morning I was doing some testing with an API while on a conference call and the call was choppy.  I thought it might be an internet performance issue and ruled that out.  Next I brought up task manager (Win10 pro) and the CPU was pegged at 100% with postman consuming ~45% while completely idle.  Did some searching and found this ticket.  Restarted postman and guess what?  Postman v8.9.0 release notes pop up.  CPU back to normal and postman <1% after the restart from the update.  So there seems to be a common theme with the updates and high CPU utilization.

**Comment by cda-1** [comment]
I've noticed that if I leave Postman open for a long time, it will eventually saturate one CPU core.  As users above have stated, it appears to be when a new version is detected. Indeed, after closing it and re-opening it, it announced a new version.

Before closing Postman, I fired up Process Monitor and watched what it was doing. I'll include the info below on the off chance it sends you in the right direction.

Is it creating a settings file over and over again?

![image](https://user-images.githubusercontent.com/13317011/128205549-eeb4d89e-d148-4da2-8c94-dce69f803c26.png)
